src/stack.rs

```diff
@@ -1,14 +1,89 @@
-// Alex Ozdemir <aozdemir@hmc.edu> // <- Your name should replace this line!
-// Starter code for HMC's MemorySafe, week 1
+// Michael Sheely <msheely@hmc.edu>
+// Developed from the Starter code for HMC's MemorySafe, week 1
 //
 // The implementation of SinglyLinkedList
 
 use Stack;
 
 pub struct SinglyLinkedList<T> {
-    data: T // Probably not what you want
+    first: Option<Box<Node<T>>>,
+    size: usize,
+}
+
+struct Node<T> {
+    payload: T,
+    next: Option<Box<Node<T>>>,
 }
 
 impl<T: Eq> Stack<T> for SinglyLinkedList<T> {
-    // ????
+    fn new() -> Self {
+        SinglyLinkedList{first: None, size: 0}
+    }
+
+    fn push_front(&mut self, item: T) {
+        self.first = Some(Box::new(Node{payload: item, next: self.first.take()}));
+        self.size += 1;
+    }
+
+    fn pop_front(&mut self) -> Option<T> {
+        if self.size > 0 {self.size -= 1;}
+        self.first.take().map(|mut node| {self.first = node.next.take(); node.payload})
+    }
+    
+    fn peek_front(&self) -> Option<&T> {
+        match self.first {
+            Some(ref node) => Some(&node.payload),
+            None => None,
+        }
+    }
+
+    fn len(&self) -> usize { self.size }
+
+    fn reverse(&mut self) {
+        use std::mem;
+        let mut current = self.first.take();
+        let mut prev = None;
+        loop {
+            // following is now current's next (taking ownership)
+            // and current now points to prev
+            let following = current.as_mut().and_then(
+                |ref mut node| mem::replace(&mut node.next, prev));
+            if following.is_none() { break; }
+            // update for next iteration of loop
+            prev = current;
+            current = following;
+        }
+        self.first = current;
+    }
```

Comment:
Nice traversal! There's another way to do this that's simpler, but perhaps less cool (see the sample solution).